### DISCUSSION TITLE
Add option to keep Instrument Name the same, regardless of Instrument Changes

### DISCUSSION THREAD

**Discussion by GabeS573** [musescore/MuseScore#11996] [Ideas]
So let's say I start off by creating a score with four percussion parts.
![Screenshot 2022-06-12 143011](https://user-images.githubusercontent.com/91428485/173254459-5b112c7a-bd47-4e2a-8065-58776cddccad.jpg)

Now, let's say I want to change the instrument names so that they display as "Percussion 1," "Percussion 2," etc. rather than "Marimba" or "Vibraphone" or "Xylophone."
![Screenshot 2022-06-12 143243](https://user-images.githubusercontent.com/91428485/173254543-c9668410-a2bd-4ea1-ba55-71c50b40f5fd.jpg)

But, now let's say that later in the score, I want the Percussion 3 (Vibraphone) player to switch to Drumset. I add an instrument change:
![image](https://user-images.githubusercontent.com/91428485/173254604-b41c4fc7-85ba-4f7f-802a-19290b6862a1.png)

However, changing the instrument in the Percussion 3 part causes the instrument name to reset on the next new line:
![image](https://user-images.githubusercontent.com/91428485/173254714-84c18546-ea2a-4c11-931a-77f7b01692e8.png)

And now the instrument name must manually be changed _again_.

At first this doesn't seem like such a big problem, but in Percussion scores with many instrument changes (i.e. 20 or more), it becomes tedious to repeatedly change the instrument names back to what they were before.

**My suggestion** is to add an option to the Staff/Part Properties menu that, if enabled for that instrument, prevents instrument changes from changing the Instrument Name for that part.

As mentioned above, this is mainly useful in percussion ensemble scores that often require a player to switch between many instruments over the course of a work. Another application (which someone brought up on Discord) is in some Jazz Band scores, where, for example, the "Alto Sax 1" player may switch to a Soprano Sax mid-song and still be labeled as "Alto Sax 1."

**Comment by cbjeukendrup** [comment]
MuseScore already sees the "part name" (i.e. player name) and "instrument names" as different concepts. Both have their own field in the Staff/Part properties dialog. 

So, I think a good solution for this problem is that we allow to choose whether the _part name_ or the _instrument name_ should be used for the staff label. So that you could just set the part name to "Percussion 3", and that will be shown as the staff label throughout the whole score, regardless of the name of the instrument playing at that time. 

(Note: with "part name", I'm **not** referring to the thing that you set in the Parts dialog. The terminology is quite confusing, see #10128.)

**Reply by GabeS573** [reply]
That would be a good solution to the problem. However, when the Instrument Name is set in the Staff/Part Properties menu, there are separate boxes for the long name and the short name (in my example, the long name was "Percussion X" and the short name was "PX"). Since there is only one Part Name currently, users would also need a way to set that "short name" and have it be consistent through the score.

(Also, after looking at the link you posted, it seems that the instrument changes in MU4 may not work exactly the way they did in MU3... so I'll have to experiment with this more.)

**Comment by aige02** [comment]
The relationship of *instruments* to *part names* has changed from MU 3 to MU 4.
In MU 3, each *instrument* had a three names and a current **instrument**.  When creating parts, the *part name* was shown in the Parts dialog when creating the **part names** to be displayed in the tab and on the part.  The **part name** could also be changed in the Parts dialog.
In MU 4, the *part name* has gone away.  Rather, the **part name** can be edited in the Parts dialog.  The *long name* is used as the default, which may cause (very minor) problems because the *long name* can be multi-line and, currently, the *part name* cannot.
I've worked around the problem by saving and restoring the long name around an instrument change.  The short name can usually be restored from the long name.  However, adding a flag, or a flag to each, should be a simple solution.  The flag(s) would default to the current behavior.  If a flag is in the other (don't change) state, then an instrument change would not affect the associated name(s).  The flag(s) should be visible in the instrument change dialog as toggles.  Also, changing the long or short name would set the associated flag to "don't change".
